### PR TITLE
Fix sentry paragraph for jump instructions

### PR DESCRIPTION
**Pull request by mayyamal**

### CODE REVIEW

File: src/cap-description.adoc [hunk 1]
```diff
@@ -279,11 +279,11 @@ For code capabilities, the sealing bit is used to implement immutable
...
 capabilities are known as sealed entry (sentry) capabilities. A program may jump to a
-sentry capability to begin executing the instructions it references. The jump
-instruction automatically unseals the capability and installs it to the
+sentry capability to begin executing the instructions it references. The <<JALR>>
+instruction automatically unseals the target capability and installs it to the
 program counter capability (see
-xref:section_riscv_programmers_model[xrefstyle=short]). The <<JALR>> instruction
-also seals the return address capability (if any) since it is the entry point
+xref:section_riscv_programmers_model[xrefstyle=short]). The jump instructions
+also seal the return address capability (if any) since it is the entry point
```
Comment (arichardson) [inline src/cap-description.adoc:286]: I'm not sure about this sentence. Maybe something more along the lines of "to ensure that the caller can only jump back to the correct address and cannot load/store via this capability".

**Review by arichardson — COMMENTED** [review]
Change looks good to me but might be worth clarifying the wording at the same time

File: src/cap-description.adoc (hunk 1 above)
Comment (mayyamal) [reply src/cap-description.adoc]: Can you take a look now? I fixed some wording in another paragraph.

File: src/cap-description.adoc [hunk 2]
```diff
@@ -280,15 +280,13 @@ it via a superset capability with <<CBLD>>. {cheri_base_ext_name} does not offer
...
 For code capabilities, the sealing bit is used to implement immutable
-capabilities that describe function entry points. Such capabilities can be leveraged
-to establish a form of control-flow integrity between mutually distrusting code. These
-capabilities are known as sealed entry (sentry) capabilities. A program may jump to a
-sentry capability to begin executing the instructions it references. The jump
-instruction automatically unseals the capability and installs it to the
+capabilities that describe function entry points, known as sealed entry (sentry) capabilities. Such capabilities can be leveraged
+to establish a form of control-flow integrity between mutually distrusting code. A program may jump to a
+sentry capability to begin executing the instructions it references. The <<JALR>>
+instruction automatically unseals a sentry target capability with zero offset and installs it to the
```
Comment (andresag01) [inline src/cap-description.adoc:286]: I would turn around this sentence to make it a little clearer:
> A <<JALR>> instruction with zero offset automatically unseal a sentry target capability and installs it to the program counter...

File: src/riscv-integration.adoc [hunk 3]
```diff
@@ -242,13 +242,13 @@ to the base behaviour as described below.
...
+instruction following the jump (<<pcc>> + 4) is sealed and written to a *c* register.
 
 <<JALR>>
-allows unconditional jumps to a target capability. The target capability is
+allows unconditional, indirect jumps to a target capability. The target capability is
 obtained by incrementing the capability in the *c* register operand by the
-sign-extended 12-bit immediate if the immediate is not zero, then setting the
-least significant bit of the result to zero, then unsealing. The capability
+sign-extended 12-bit offset, then setting the
+least significant bit of the result to zero. The target capability is unsealed if it is a sentry with offset zero. The capability
```
Comment (tariqkurd-repo) [inline src/riscv-integration.adoc:251]:
```suggestion
least significant bit of the result to zero. The target capability is unsealed if it is a sentry with zero offset. The capability
```

File: src/cap-description.adoc [hunk 4]
```diff
@@ -280,15 +280,13 @@ it via a superset capability with <<CBLD>>. {cheri_base_ext_name} does not offer
...
 program counter capability (see
-xref:section_riscv_programmers_model[xrefstyle=short]). The <<JALR>> instruction
-also seals the return address capability (if any) since it is the entry point
-to the caller function.
+xref:section_riscv_programmers_model[xrefstyle=short]). The jump instructions
+also seal the return address capability which serves as an entry point the caller can jump back to but cannot use to authorize memory loads or stores.
```
Comment (tariqkurd-repo) [inline src/cap-description.adoc:289]:
```suggestion
also seal the return address capability which serves as an entry point the callee can jump back to but cannot use to authorize memory loads or stores.
```

File: src/cap-description.adoc [hunk 5]
```diff
@@ -280,15 +280,13 @@ it via a superset capability with <<CBLD>>. {cheri_base_ext_name} does not offer
...
 
 For code capabilities, the sealing bit is used to implement immutable
-capabilities that describe function entry points. Such capabilities can be leveraged
-to establish a form of control-flow integrity between mutually distrusting code. These
-capabilities are known as sealed entry (sentry) capabilities. A program may jump to a
-sentry capability to begin executing the instructions it references. The jump
-instruction automatically unseals the capability and installs it to the
+capabilities that describe function entry points, known as sealed entry (sentry) capabilities. Such capabilities can be leveraged
+to establish a form of control-flow integrity between mutually distrusting code. A program may jump to a
+sentry capability to begin executing the instructions it references. The <<JALR>>
```
Comment (tariqkurd-repo) [inline src/cap-description.adoc:285]:
```suggestion
sentry capability to begin executing the instructions it references. A <<JALR>>
```

File: src/cap-description.adoc (hunk 2 above)
Comment (tariqkurd-repo) [inline src/cap-description.adoc:286]:
```suggestion
instruction with zero offset automatically unseals a sentry target capability and installs it to the
```

File: src/cap-description.adoc [hunk 6]
```diff
@@ -280,15 +280,13 @@ it via a superset capability with <<CBLD>>. {cheri_base_ext_name} does not offer
...
-instruction automatically unseals the capability and installs it to the
+capabilities that describe function entry points, known as sealed entry (sentry) capabilities. Such capabilities can be leveraged
+to establish a form of control-flow integrity between mutually distrusting code. A program may jump to a
+sentry capability to begin executing the instructions it references. A <<JALR>>
+instruction with zero offset automatically unseals a sentry target capability and installs it in the
 program counter capability (see
-xref:section_riscv_programmers_model[xrefstyle=short]). The <<JALR>> instruction
-also seals the return address capability (if any) since it is the entry point
-to the caller function.
+xref:section_riscv_programmers_model[xrefstyle=short]). The jump instructions
```
Comment (tariqkurd-repo) [inline src/cap-description.adoc:288]:
```suggestion
xref:section_riscv_programmers_model[xrefstyle=short]). The jump=and-link instructions
```

**Review by tariqkurd-repo — APPROVED** [review]

File: src/cap-description.adoc (hunk 2 above)
Comment (tariqkurd-repo) [reply src/cap-description.adoc]: done

File: src/cap-description.adoc (hunk 1 above)
Comment (tariqkurd-repo) [reply src/cap-description.adoc]: I've made some updates

**Pull request merged**